internal/cloudapi/v2/handler.go

```diff
@@ -1178,3 +1186,183 @@ func genRepoConfig(repo Repository) (*rpmmd.RepoConfig, error) {
 
 	return repoConfig, nil
 }
+
+func (h *apiHandlers) PostImageFromCompose(ctx echo.Context, id string) error {
+	return h.server.EnsureJobChannel(h.postImageFromComposeImpl)(ctx, id)
+}
+
+func (h *apiHandlers) postImageFromComposeImpl(ctx echo.Context, id string) error {
+	channel, err := h.server.getTenantChannel(ctx)
+	if err != nil {
+		return HTTPErrorWithInternal(ErrorTenantNotFound, err)
+	}
+
+	jobId, err := uuid.Parse(id)
+	if err != nil {
+		return HTTPError(ErrorInvalidComposeId)
+	}
+
+	jobType, err := h.server.workers.JobType(jobId)
+	if err != nil {
+		return HTTPError(ErrorComposeNotFound)
+	}
+
+	if jobType != "osbuild" {
+		return HTTPError(ErrorInvalidJobType)
+	}
+
+	var osbuildResult worker.OSBuildJobResult
+	osbuildStatus, deps, err := h.server.workers.OSBuildJobStatus(jobId, &osbuildResult)
+	if err != nil {
+		return HTTPErrorWithInternal(ErrorGettingOSBuildJobStatus, err)
+	}
+
+	if osbuildStatus.Finished.IsZero() || !osbuildResult.Success {
+		return HTTPError(ErrorComposeBadState)
+	}
+
+	us, err := uploadStatusFromOSBuildJobResult(&osbuildResult)
+	if err != nil {
+		return err
+	}
+
+	// the id of the last job in the dependency chain which users should wait on
+	finalJob := jobId
+	// look at the upload status of the osbuild dependency to decide what to do
+	switch us.Type {
+	case UploadTypesAws:
+		options := us.Options.(AWSEC2UploadStatus)
+		var img AWSEC2ImageFromCompose
+		err := ctx.Bind(&img)
+		if err != nil {
+			return err
+		}
+
+		shareAmi := options.Ami
+		shareRegion := img.Region
+		if img.Region != options.Region {
+			// Let the share job use dynArgs
+			shareAmi = ""
+			shareRegion = ""
+
+			// Check deps if we need to do a copyjob
+			foundDep := false
+			for _, d := range deps {
```

Comment:
`deps` are dependencies of the osbuild job from above, not jobs which have the osbuild job as a dependency (I mean unless I'm heavily confused).

We never enqueue osbuild job with the ec2-copy job as its dependency and we also don't modify dependencies of the original osbuild job.

So my question is whether it makes any sense to check the osbuild job dependencies? 🤔 

Personally I think that the ec2-copy dependency can not be ever found and the code path will always follow into 
```go
if !foundDep {
   ...
}
```

Reply:
Ugh you're right, this is wrong. I'll need to add something which queries dependents rather.